OpenRA.Mods.Common/Util.cs

```diff
@@ -155,5 +158,70 @@ public static IEnumerable<CPos> RandomWalk(CPos p, MersenneTwister r)
 				yield return p;
 			}
 		}
+
+		public static string FriendlyTypeName(Type t)
+		{
+			if (t.IsGenericType && t.GetGenericTypeDefinition() == typeof(HashSet<>))
+				return "Set of {0}".F(t.GetGenericArguments().Select(FriendlyTypeName).ToArray());
+
+			if (t.IsGenericType && t.GetGenericTypeDefinition() == typeof(Dictionary<,>))
+				return "Dictionary<{0},{1}>".F(t.GetGenericArguments().Select(FriendlyTypeName).ToArray());
+
+			if (t.IsGenericType && t.GetGenericTypeDefinition() == typeof(IList<>)) // TODO: why does this not work?
+				return "List of {0}".F(FriendlyTypeName(t.GetElementType()));
+
+			if (t.IsSubclassOf(typeof(Array)))
+				return "Multiple {0}".F(FriendlyTypeName(t.GetElementType()));
+
+			if (t.IsGenericType && t.GetGenericTypeDefinition() == typeof(Primitives.Cache<,>))
+				return "Cached<{0},{1}>".F(t.GetGenericArguments().Select(FriendlyTypeName).ToArray());
+
+			if (t == typeof(int) || t == typeof(uint))
+				return "Integer";
+
+			if (t == typeof(int2))
+				return "2D Integer";
+
+			if (t == typeof(float) || t == typeof(decimal))
+				return "Real Number";
+
+			if (t == typeof(float2))
+				return "2D Real Number";
+
+			if (t == typeof(CPos))
+				return "2D Cell Position";
+
+			if (t == typeof(CVec))
+				return "2D Cell Vector";
+
+			if (t == typeof(WAngle))
+				return "1D World Angle";
+
+			if (t == typeof(WRot))
+				return "3D World Rotation";
+
+			if (t == typeof(WPos))
+				return "3D World Position";
+
+			if (t == typeof(WDist))
+				return "1D World Distance";
+
+			if (t == typeof(WVec))
+				return "3D World Vector";
+
+			if (t == typeof(HSLColor))
+				return "hue, saturation, lightness Color";
```

Comment:
It should be combined with the `Color` case and say something like `Color (RRGGBB[AA] notation)`.